src/script/sign.cpp

```diff
@@ -355,3 +355,32 @@ bool DummySignatureCreator::CreateSig(std::vector<unsigned char> &vchSig,
     vchSig[6 + 33 + 32] = SIGHASH_ALL;
     return true;
 }
+
+
+std::vector<unsigned char> signmessage(const std::vector<unsigned char> &data, const CKey &key)
+{
+    CHashWriter ss(SER_GETHASH, 0);
+    ss << strMessageMagic << data;
+
+    vector<unsigned char> vchSig;
+    if (!key.SignCompact(ss.GetHash(), vchSig)) // signing will only fail if the key is bogus
+    {
+        DbgAssert(!"bad key", );
+        return std::vector<unsigned char>();
+    }
+    return vchSig;
+}
+
+std::vector<unsigned char> signmessage(const std::string &data, const CKey &key)
+{
```

Comment:
I wanted to avoid the semantically meaningless, but inefficient conversion.  A template would be possible.  In this case I wanted to keep things simple since the function is only 5 lines.

Reply:
agree with @dagurval , i think a template would be appropriate here.